components/ApiKeyUsageChart.js

```diff
@@ -0,0 +1,193 @@
+            })
+          }
+        }
+      })
+    } else {
+      requestData = plan.result.items[id]
+    }
+    return requestData
+  }
+
+  _getApiUser = () => this.props.aggregatedView
+    ? null
+    : this.props.plan.apiUsers[this.props.id]
+
+  _setValue = (data, event) => {
+    this.setState({value: data})
+  }
+
+  _viewApiKey = () => {
+    const { id, plan } = this.props
+    const userForKey = plan.apiUsers[id]
+    const key = userForKey.apiKeys.find(key => key.keyId === id)
+    window.prompt('Copy and paste the API key to use in requests', key.value)
+  }
+
+  render () {
+    const { aggregatedView, id, plan } = this.props
+    const { value } = this.state
+    const ONE_DAY_MILLIS = 86400000
+    const startDate = moment(plan.result.startDate)
+    if (!aggregatedView && !id) {
+      console.warn('Cannot show non-aggregated view if id prop is undefined.')
+      return null
+    }
+    const apiUser = this._getApiUser()
+    // Render the # of requests per API key on each day beginning
+    // with the start date.
+    const requestData = this._getRequestData()
+    const timestamp = startDate.valueOf()
+    let rangeMax = 0
+    // Format request data for chart component.
+    const CHART_DATA = requestData
+      .map((value, i) => {
+        if (i > 0) startDate.add(1, 'days')
+        const begin = startDate.valueOf()
+        const y = value[0]
+        const end = begin + ONE_DAY_MILLIS
+        if (y > rangeMax) rangeMax = y
+        return ({x0: begin, x: end, y})
+      })
+    const maxY = rangeMax === 0 ? 10 : Math.ceil(rangeMax / 10) * 10
+    return (
+      <div className='usage-list' style={{display: 'inline-block'}}>
+        {this._getChartTitle()}
+        <p>
+          {id && <span><Key size={20} style={{marginRight: 10}} />{id}</span>}
+          {apiUser &&
+            <small>
+              <Button
+                onClick={this._viewApiKey}
+                size='sm'
+                variant='link'
+              >
+                click to view key
+              </Button>
+            </small>
+          }
+        </p>
+        <XYPlot
+          xDomain={[timestamp - 2 * ONE_DAY_MILLIS, timestamp + 30 * ONE_DAY_MILLIS]}
+          // Round up max y value to the nearest 10
+          yDomain={[0, maxY]}
+          xType='time'
+          width={300}
+          height={300}
+        >
+          <VerticalGridLines />
+          <HorizontalGridLines />
+          <XAxis />
+          <YAxis />
+          <VerticalRectSeries
+            data={CHART_DATA}
+            // Update value on mouse over/out.
+            onValueMouseOut={this._clearValue}
+            onValueMouseOver={this._setValue}
+            style={{stroke: '#fff'}} />
+        </XYPlot>
+        <p style={{textAlign: 'center'}}>
+          {value
+            ? <>
+              {moment(value.x).format('MMM DD')}: {value.y} requests
+            </>
+            : <>
+              [Hover over bars to see values.]
```

Comment:
Suggestion: move this hint text in a paragraph above the charts?

Reply:
I think it makes more sense to keep it below. My cursor tends to be located at the baseline of the chart as I hover over the bars, so I would prefer to see the data close to the cursor.